### PR TITLE
Feature/polls email and sms

### PR DESCRIPTION
**Pull request by Wardormeur**

### CODE REVIEW

File: lib/poll/channels/send-sms-poll.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    function (err, generated) {
+      // If there is an error while generating the template, it's going to crash watsoever because we can't catch it
+      wfCb(null, generated.text);
+    });
+  };
+
+  var sendSMS = function (template, wfCb) {
+    client.sendMessage({
+        // We need to append a plus or twilio will consider the area number
+        // is the one from the purchased number (appending +252 in our case)
```
Comment (DanielBrierton) [inline lib/poll/channels/send-sms-poll.js]: You mean +353 😛

File: lib/poll/channels/send-sms-poll.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        to: '+' + sms.phoneNumber,
+        // TODO : ? retrieve from twilio API https://www.twilio.com/docs/api/rest/incoming-phone-numbers#instance ?
+        // Still needs an ID which is relative to the phone number, pointless ?
+        from: process.env.TWILIO_PHONE_NB,
+        body: template
+    }, function (error, message) {
+        if (error) {
+          console.error('Oops! There was an error.', error);
```
Comment (DanielBrierton) [inline lib/poll/channels/send-sms-poll.js]: Good to know 😛

File: lib/poll/start-poller.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+   var limit = args.limit || 200;
+
+   var updatePoll = function (wfCb) {
+     seneca.act({role: plugin, cmd: 'save_poll_setup', poll: {id: pollId, startedAt: new Date()}}, function (err){
+       if (err) return done(err);
+       return wfCb(null);
+     });
+   };
+
+   var enqueue = function (wfCb) {
```
Comment (DanielBrierton) [inline lib/poll/start-poller.js]: Take a look at count/limit thing...

File: lib/poll/channels/get-twiml.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ *  tpl: '',
+ *   content: {}
+ * }
+ * @return {String}            Stringified twiML
+ */
+function getTwiML (args, callback) {
+  var seneca = this;
+  var plugin = args.role;
+  var config = args.data;
+  seneca.act({role: 'mail', cmd: 'generate', code: args.data.tpl, content: args.data.content },
```
Comment (Wardormeur) [inline lib/poll/channels/get-twiml.js]: i didn't feel like creating seneca-twilio :p

File: lib/poll/channels/queue-sms-poll.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  var setTemplate = function (champs, waterfallCb) {
+    var payloads = {};
+    async.mapSeries(champs, function (champ, next) {
+      if (payloads[champ.id]) {
+        payloads[champ.id].template = 'polls-sms-notification-reminder-';
+      } else {
+        payloads[champ.id] = champ;
+        payloads[champ.id].template = 'polls-sms-notification-';
+      }
+      var locality = champ.alpha2 ? champ.alpha2.toLowerCase() + '_' + champ.alpha2 : 'en_US';
```
Comment (Wardormeur) [inline lib/poll/channels/queue-sms-poll.js]: That's what we're using right now for emails :p

File: lib/poll/channels/save-sms-result.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    seneca.act({role: 'cd-profiles', cmd: 'list', query: {phone: phone}},
+     function (err, champProfiles) {
+      if (err) return done(err);
+      seneca.act({role: plugin, cmd: 'load_usersdojos', query: {userId: {in$: _.map(champProfiles, 'userId')}}},
+       function (err, usersDojos) {
+        if (err) return done(err);
+        var champs = _.filter(usersDojos, function (user) {
+          return user.userTypes.indexOf('champion') > -1;
+        });
+        // if (champs.length > 1) {
```
Comment (Wardormeur) [inline lib/poll/channels/save-sms-result.js]: Oupsies, leftovers of yesterday tests :p

File: lib/poll/channels/save-sms-result.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+  };
+
+  var savePollResult = function (sms, poll, champProfile, waterfallCb) {
+    var result = {
+      value: sms.Body,
+      dojoId: champProfile.dojoId,
+      pollId: poll.id
+    };
+
+    seneca.act({role: plugin, cmd: 'save_poll_result', poll: result},
```
Comment (Wardormeur) [inline lib/poll/channels/save-sms-result.js]: It's prone to error : say somebody respond "on the 20th, we had 10 kids", we would not be able to know what to store. I prefer an erroneous content that is saved "as it" than a best-effort that may be faulty, even more considering we have the f-end to handle those erroneous behaviors :)

File: lib/poll/channels/send-sms-poll.js (hunk 2 above)
Comment (Wardormeur) [inline lib/poll/channels/send-sms-poll.js]: Better sorry than dead !

File: lib/poll/channels/save-sms-result.js (hunk 7 above)
Comment (DanielBrierton) [inline lib/poll/channels/save-sms-result.js]: Fair enough

File: lib/poll/channels/save-sms-result.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    if (!fs.existsSync(path.join(__dirname, '/email-templates/', templateCode))) locality = 'en_US';
+    seneca.act({role: plugin, cmd: 'get_twiML', data: {
+      tpl: templateCode +  locality,
+      content: {
+        pollLink: protocol + '://' + zenHostname + '/dashboard/poll/' + poll.id + '/dojo/' + champProfile.dojoId
+      }
+    }},
+     function (err, twiML) {
+      if (err) return done(err);
+      console.log('TWIML', twiML);
```
Comment (butlerx) [inline lib/poll/channels/save-sms-result.js]: console?

**Pull request merged**